Transformations/Models.MetaTransformation/Meta/Class2Type.cs

```diff
@@ -832,12 +830,11 @@ protected virtual CodeMemberMethod CreateGetAttributeValue(IClass input, CodeTyp
                 return getAttributeValue;
             }
 
-            private CodeMemberMethod AddToGetAttributeValue(CodeMemberMethod method, IAttribute attribute, ITransformationContext context)
+            private CodeMemberMethod AddToGetAttributeValue(CodeMemberMethod method, IAttribute attribute, CodeMemberProperty property, ITransformationContext context)
             {
                 if (attribute.UpperBound == 1 || attribute.IsOrdered)
                 {
-                    var attributeProperty = context.Trace.ResolveIn(Rule<Attribute2Property>(), attribute);
-                    var propRef = new CodePropertyReferenceExpression(new CodeThisReferenceExpression(), attributeProperty.Name);
+                    var propRef = new CodePropertyReferenceExpression(new CodeThisReferenceExpression(), property.Name);
```

Comment:
The attribute property here is a good source of truth, but I agree you can pass the real property in as a parameter as well, makes the tracing superfluous.